src/EconomicStrategy/EconomicStrategyManager.ts

```diff
@@ -237,4 +244,64 @@ export class EconomicStrategyManager {
 
     return true;
   }
+
+  // tslint:disable-next-line:cognitive-complexity
+  private async smartGasEstimation(txRequest: ITxRequest): Promise<BigNumber | null> {
+    const normalizeWaitTimes = (stats: any) => {
+      let normalizedTimes = {
+        safeLow: stats.safeLowWait.mul(10),
+        avg: stats.avgWait.mul(10),
+        fast: stats.fastWait.mul(10),
+        fastest: stats.fastestWait.mul(10)
+      };
+
+      if (txRequest.temporalUnit === 1) {
+        // Normalize the estimate
+        const blockTime = stats.block_time;
+        normalizedTimes = {
+          safeLow: Math.floor(normalizedTimes.safeLow.div(blockTime)),
+          avg: Math.floor(normalizedTimes.avg.div(blockTime)),
+          fast: Math.floor(normalizedTimes.fast.div(blockTime)),
+          fastest: Math.floor(normalizedTimes.fastest.div(blockTime))
+        };
+      }
+
+      return normalizedTimes;
+    };
+
+    const gasStats = await W3Util.getEthGasStationStats();
+    if (gasStats) {
```

Comment:
Inverse the if and return null 

if (!gasStats) {
  return null;
}

This lower the cognitive-complexity.